distribution/client.go

```diff
@@ -329,6 +302,34 @@ func (c *Client) Tag(source string, target string) error {
 	return c.store.AddTags(source, []string{target})
 }
 
+// PushModel pushes a tagged model from the content store to the registry.
+func (c *Client) PushModel(ctx context.Context, tag string) (err error) {
+	// Parse the tag
+	ref, err := name.NewTag(tag)
+	if err != nil {
+		return fmt.Errorf("invalid tag %q: %w", tag, err)
+	}
+
+	// Get the model from the store
+	mdl, err := c.store.Read(tag)
```

Comment:
oh just realized that `store.Read` does not return a `ErrModelNotFound` if the model is not found. Not related to this PR, just to share

Reply:
I can fix in this PR and add some tests to ensure we are getting the proper sentinel error in both this case and the regular get case.